src/main/kotlin/org/rust/cargo/toolchain/ProxyHelper.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.cargo.toolchain
+
+import com.intellij.util.net.HttpConfigurable
+import java.net.URI
+
+class ProxyHelper(var httpSettings: HttpConfigurable) {
+
+    fun isHttpProxyEnabled(): Boolean {
+        return httpSettings.USE_HTTP_PROXY && httpSettings.PROXY_HOST.isNotEmpty()
+    }
+
+    fun getProxyUri(): URI {
+        var userInfo: String? = null
+        if (httpSettings.PROXY_AUTHENTICATION
+            && httpSettings.proxyLogin!!.isNotEmpty() && httpSettings.plainProxyPassword != null) {
```

Comment:
Hm, why are these `!!` safe? Is it because the user of this class required to call `isHttpProxyEnabled` before?

Could we fold both the checking and modification logic into this class?

Something like this:

```Rust
fun withProxyIfNeeded(cmd: GeneralCommandLine): GeneralCommandLine {
    if (!proxyEnabled) return cmd
    ....
}
```